### PR TITLE
Update Envoy Proxy versions and add new metrics to envyproxy metricbeat module

### PR DESCRIPTION
**Pull request by berfinsari**
This PR fixes elastic/beats#13642 and adding new metrics to server metricset.

### CODE REVIEW

File: metricbeat/docker-compose.yml [hunk 1]
```diff
@@ -103,11 +103,11 @@ services:
       - 9200
 
   envoyproxy:
-    image: docker.elastic.co/observability-ci/beats-integration-envoyproxy:${ENVOYPROXY_VERSION:-v1.7.0}-1
+    image: docker.elastic.co/observability-ci/beats-integration-envoyproxy:${ENVOYPROXY_VERSION:-v1.12.0}-1
```
Comment (ChrsMark) [inline metricbeat/docker-compose.yml]: I think we should keep testing against both the old and the newest version, like what we do in other modules like [redis](https://github.com/elastic/beats/blob/fc3b51a864533d8100814dab33f75825be16284a/metricbeat/tests/system/test_redis.py#L164)

**Comment by jsoriano** [comment]
ok to test

**Comment by jsoriano** [comment]
jenkins, test this again please

**Comment by jsoriano** [comment]
@berfinsari sorry, one last thing, could you add a changelog entry?

**Comment by berfinsari** [comment]
Thanks for the review! @jsoriano

**Comment by jsoriano** [comment]
jenkins, test this again please

**Comment by jsoriano** [comment]
Merged and backport created, this will be hopefuly released in 7.6.0. Thanks @berfinsari!

**Pull request merged**